sql/mysqld.cc

```diff
@@ -2084,16 +2089,31 @@ static void clean_up(bool print_message)
 */
 static void wait_for_signal_thread_to_end()
 {
-  uint i;
+  uint i, n_waits= DBUG_EVALUATE("force_sighup_processing_timeout", 5, 100);
+  int err;
```

Comment:
should be initialized `int err=ESRCH;`  or so.  Now fails on Windows with uninitialized variable `err` in line 2106 . The variable is not initialized, because `signal_thread_in_use` is never set, and loop body never runs.  That's because Windows does not use signal_hand thread,  entire shutdown sequence runs in the main thread (there is also no sigwait or SIGHUP here either). So this function should be no-op on Windows, but it could also be no-op if signal thread did yet start, who knows.